pkg/rpc/server.go

```diff
@@ -224,6 +233,10 @@ func (s *Server) init() error {
 	if s.enabelGRPCReflection {
 		reflection.Register(s.grpcServer)
 	}
+	// NOTE: This should be registered after all services have been registered.
+	if s.prometheusUnaryInterceptor != nil {
```

Comment:
We have to add `s.prometheusUnaryInteceptor` into `unaryInterceptors` in case it is not nil?

Reply:
Yes we should. Good catch!